cms/0000-TinyMCE-RTE-Improvements.md

```diff
@@ -0,0 +1,81 @@
+# RFC Name
+
+Request for Contribution (RFC) 0000 : TinyMCE RTE Improvements
+
+## Code of conduct
+
+Please read and respect the [RFC Code of Conduct](https://github.com/umbraco/rfcs/blob/master/CODE_OF_CONDUCT.md)
+
+## Intended Audience
+
+The intended audience for this RFC is:
+* Developers & implementors of Umbraco
+
+## Summary
+
+This RFC is to give specific details and implmentations on how we plan to improve the existing implementation of the TinyMCE version 4 rich text editor.
+
+
+## Motivation
+This has been a follow up RFC from #5 & #15 where it has been discusssed that the rich text editor could be improved and enhanced to help give content editors a better experience when authoring or ammending content in the CMS.
+
+
+## Detailed Design
+
+The following improvements to the TinyMCE editor are planned and in another round we can do further iterative improvements.
+
+**Implement pasting along with drag n drop of images**<br/>
+We have verified with a CodePen sample to ensure we could get TinyMCE to support this. We will allow a prevalue configuration of the editor in Umbraco to set a folder location of where to store pasted and dragged images in the media library. A future iteration could improve this further and prompt the user on where they would like to store the image/s in the media library.
+
+**Reduce inital load time and file size of TinyMCE into the browser**<br/>
+Utilize Client Dependancy Framework that is shipped with Umbraco to help reduce the initial file size.
+
+**Improve the loading experience** <br/>
+Use Tiny's `init_inistance_callback` property to remove Umbraco's own loader as opposed to the current implementation of checking it the Tiny library is loaded.
+
+**Improve the experience with pasting from Word documents**<br/>
+Use Tiny's event `before_paste` to notify us when content is being pasted into the editor. This event already notifies us if the content was copied from Word, so we can use the pasted content to some JavaScript open source library or C# library to help clean up the HTML markup & formatting.
+
```

Comment:
We should add:

**Improve moving un-editable elements**

For non-editable elements inserted into the rich text editor (such as macros, or OEmbed content) we want to allow for moving these elements in a nicer way by allowing copy/pasting or moving of these elements (by holding shift when they are highlighted + arrow keys).